java/org/contikios/cooja/Simulation.java

```diff
@@ -53,13 +51,27 @@
  *
  * @author Fredrik Osterlind
  */
-public class Simulation extends Observable implements Runnable {
+public class Simulation extends Observable {
+
+  /* Internal marker for notifying simulation thread to start running simulation */
+  private static final Runnable START = () -> {};
```

Comment:
It's hopeless to print functions when debugging, make these integers instead. The switch in Java 17 can match on object types (`Integer i`, `null`, and I think `Runnable`).

Reply:
The idea was to have the blocking queue typed and only contain Runnable. Changing this would mean using Object in the blocking queue and use instanceof or switch pattern matching. I can change this, but the queue will still contain Runnable since it is used by invokeSimulationThread().

Reply:
Yes, I think that will be easier to deal with in the long run since most elements can be printed. (The type signature of invokeSimulationThread() ensures that only Runnables are added to the queue from the outside.)

Reply:
I have updated this now.